tests/openfl/utilities/test_dump_requirements.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2020-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Dump requirements file test module."""
+
+from pathlib import Path
+
+import pytest
+from pip._internal.operations import freeze
+
+from openfl.utilities.workspace import dump_requirements_file
+
+LINES = ['-o option\n', 'package==0.0.1\n']
+PREFIXES = ('-u test',)
+PREFIXES_OVERLAP = ('-u test', '-o option')
+
+
+@pytest.fixture
+def requirements_file():
+    """Prepare test requirements file."""
+    path = Path('./test_requirements.txt').absolute()
+    with open(path, 'w') as f:
+        f.writelines(LINES)
+    yield path
+    path.unlink()
+
+
+@pytest.mark.parametrize('keep_original_prefixes,prefixes,expected_lines', [
+    (False, None, LINES[1:]),
+    (True, None, LINES),
+    (False, PREFIXES, [p + '\n' for p in PREFIXES] + [li for li in LINES if li[0] != '-']),
+    (True, PREFIXES, [li for li in LINES if li[0] == '-'
+                      ] + [p + '\n' for p in PREFIXES] + [li for li in LINES if li[0] != '-']),
+    (True, PREFIXES_OVERLAP, [p + '\n' for p in PREFIXES_OVERLAP
+                              ] + [li for li in LINES if li[0] != '-']),
+])
```

Comment:
To simplify it, maybe make sense:
@pytest.mark.parametrize('keep_original_prefixes,prefixes,expected_lines', [
    (False, None, REQ_WITHOUT_PREFIXES),
    (True, None, REQ_WITH_PREFIXES),
    (False, PREFIXES, PREFIXES + REQ_WITHOUT_PREFIXES,
    (True, PREFIXES, PRESIXES + REQ_WITH_PREFIXES,
    (True, PREFIXES_OVERLAP, PREFIXES_OVERLAP + REQ_WITH_PREFIXES),
])
Because multiple list comprehension hard to read.
Also maybe this test should be separated to increase readability.

Reply:
These list comprehension will be hard to read but in another place

Reply:
Replaced list comprehension with variables